### PR TITLE
Support GM edge cluster/node data sources

### PR DESCRIPTION
**Pull request by asarfaty**

### CODE REVIEW

File: nsxt/data_source_nsxt_policy_edge_cluster.go [hunk 1]
```diff
@@ -20,17 +22,55 @@ func dataSourceNsxtPolicyEdgeCluster() *schema.Resource {
 			"display_name": getDataSourceDisplayNameSchema(),
 			"description":  getDataSourceDescriptionSchema(),
 			"path":         getPathSchema(),
+			"site_path": {
+				Type:         schema.TypeString,
+				Description:  "Path of the site this Transport Zone belongs to",
```
Comment (annakhm) [inline nsxt/data_source_nsxt_policy_edge_cluster.go:27]: Edge Cluster

File: nsxt/data_source_nsxt_policy_edge_cluster.go [hunk 2]
```diff
@@ -20,17 +22,55 @@ func dataSourceNsxtPolicyEdgeCluster() *schema.Resource {
...
+		if err != nil {
+			return err
+		}
+		converter := bindings.NewTypeConverter()
+		converter.SetMode(bindings.REST)
+		dataValue, errors := converter.ConvertToGolang(obj, gm_model.PolicyEdgeClusterBindingType())
+		if len(errors) > 0 {
+			return errors[0]
+		}
+		edgeClusterResource := dataValue.(gm_model.PolicyEdgeCluster)
```
Comment (annakhm) [inline nsxt/data_source_nsxt_policy_edge_cluster.go:63]: All these are standard attributes, so these are already set in policyDataSourceResourceRead. Looks like the conversion to specific binding type is not needed here?

File: nsxt/data_source_nsxt_policy_edge_node.go [hunk 3]
```diff
@@ -34,17 +37,40 @@ func dataSourceNsxtPolicyEdgeNode() *schema.Resource {
...
+		query := make(map[string]string)
+		query["parent_path"] = edgeClusterPath
+		if memberIndexSet {
+			query["member_index"] = strconv.Itoa(memberIndex.(int))
+		}
+		obj, err := policyDataSourceResourceReadWithValidation(d, getPolicyConnector(m), "PolicyEdgeNode", query, false)
+		if err != nil {
+			return err
+		}
+		converter := bindings.NewTypeConverter()
```
Comment (annakhm) [inline nsxt/data_source_nsxt_policy_edge_node.go:53]: same as above

File: website/docs/d/policy_transport_zone.html.markdown [hunk 4]
```diff
@@ -26,6 +26,10 @@ data "nsxt_policy_transport_zone" "vlan_transport_zone" {
 
 Note: This usage is for Global Manager only.
 ```hcl
+data "nsxt_policy_site" "paris" {
```
Comment (enhaocui) [inline website/docs/d/policy_transport_zone.html.markdown:29]: Thanks for this catch!

File: website/docs/d/policy_edge_cluster.html.markdown [hunk 5]
```diff
@@ -17,12 +17,27 @@ data "nsxt_policy_edge_cluster" "ec" {
...
+}
+```
+
 ## Argument Reference
 
 * `id` - (Optional) The ID of the edge cluster to retrieve.
 
 * `display_name` - (Optional) The Display Name prefix of the edge cluster to retrieve.
 
+* `site_path` - (Optional) The path of the site which the Transport Zone belongs to, this configuration is required for global manager only. `path` field of the existing `nsxt_policy_site` can be used here.
```
Comment (enhaocui) [inline website/docs/d/policy_edge_cluster.html.markdown:39]: edge cluster

File: nsxt/data_source_nsxt_policy_edge_cluster.go (hunk 2 above)
Comment (enhaocui) [reply nsxt/data_source_nsxt_policy_edge_cluster.go:63]: +1

File: nsxt/data_source_nsxt_policy_edge_cluster.go (hunk 1 above)
Comment (asarfaty) [reply nsxt/data_source_nsxt_policy_edge_cluster.go:27]: Done

File: nsxt/data_source_nsxt_policy_edge_cluster.go (hunk 2 above)
Comment (asarfaty) [reply nsxt/data_source_nsxt_policy_edge_cluster.go:63]: Done

File: nsxt/data_source_nsxt_policy_edge_node.go (hunk 3 above)
Comment (asarfaty) [reply nsxt/data_source_nsxt_policy_edge_node.go:53]: Done

File: website/docs/d/policy_edge_cluster.html.markdown (hunk 5 above)
Comment (asarfaty) [reply website/docs/d/policy_edge_cluster.html.markdown:39]: Done

File: nsxt/data_source_nsxt_policy_edge_node.go [hunk 6]
```diff
@@ -55,6 +68,9 @@ func dataSourceNsxtPolicyEdgeNodeRead(d *schema.ResourceData, m interface{}) err
 		obj = objGet
 	} else {
 		// Get by full name/prefix
+		name, nameSet := d.GetOkExists("display_name")
```
Comment (annakhm) [inline nsxt/data_source_nsxt_policy_edge_node.go:71]: This should be GetOk :)

File: nsxt/data_source_nsxt_policy_edge_node.go [hunk 7]
```diff
@@ -34,17 +35,29 @@ func dataSourceNsxtPolicyEdgeNode() *schema.Resource {
 
 func dataSourceNsxtPolicyEdgeNodeRead(d *schema.ResourceData, m interface{}) error {
 	// Read an edge node by name or id
+	edgeClusterPath := d.Get("edge_cluster_path").(string)
+	memberIndex, memberIndexSet := d.GetOkExists("member_index")
```
Comment (annakhm) [inline nsxt/data_source_nsxt_policy_edge_node.go:39]: This should be GetOk - I see that GetOkExists doesn't work for non-bools

File: nsxt/data_source_nsxt_policy_edge_node.go (hunk 6 above)
Comment (asarfaty) [reply nsxt/data_source_nsxt_policy_edge_node.go:71]: Done

File: nsxt/data_source_nsxt_policy_edge_node.go (hunk 7 above)
Comment (asarfaty) [reply nsxt/data_source_nsxt_policy_edge_node.go:39]: Done

**Pull request merged**